### ISSUE TITLE
page views counter

### ISSUE THREAD

**Issue by frase-git** [cmb69/realblog_xh#69]
Die Page-Views werden in der DB pro Klick mit ID und Timestamp erfasst.
Mal abgeshen davon, dass die DB dadurch auch schon mal sehr groß werden kann, gibt es keine Möglichkeit den Couter zu beeinflussen. Oder? (außer mit DB-Browser u.Ä.)
Ich frage mich auch, was passiert mit dem Zähler bei gelöschten Artikeln? Wird die DB bereinigt?
Vielleicht wäre es besser den Zähler in der Tabelle "articles" mit anzuhängen, statt in einer extra-Tabelle zu führen. Bei jedem Klick um Eins erhöhen. Wenn der Artikel gelöscht wird, ist auch der Zähler weg.
Und schön wäre es, wenn in der Artikel-Bearbeitung die Page-Views in einem einfachen Textfeld änderbar angezeigt würden.

**Comment by cmb69** [comment]
> Ich frage mich auch, was passiert mit dem Zähler bei gelöschten Artikeln? Wird die DB bereinigt?

Nein, zur Zeit nicht. Allerdings nicht, weil das technisch ein besonderes Problem wäre, sondern wegen der CSV-Import-Export-Möglichkeit. An sich ist auch das kein wirkliches Problem, aber ich wollte die beiden Tabellen fürs erste lieber nicht zu fest koppeln. Ein wirkliches Problem sollte das aber nicht darstellen, und vermutlich muss ich sowieso noch eine Möglichkeit liefern die Datenbank zu kompaktieren (werden Datensätze gelöscht, dann wird der Speicher nicht unbedingt frei gegeben), und dann kann ich das auch mit einbauen.

> Vielleicht wäre es besser den Zähler in der Tabelle "articles" mit anzuhängen, statt in einer extra-Tabelle zu führen. Bei jedem Klick um Eins erhöhen.

Das wurde ja bereits im Forum besprochen (in issue #35 finden sich die Links). Ich fand das [Argument](https://cmsimpleforum.com/viewtopic.php?f=16&t=12420#p57523), dass die meisten entsprechenden WP Plugins auch einen Zeitraum für die "most populars" erlauben, schon recht überzeugend. Und auch wenn diese Möglichkeit noch nicht besteht, könnte sie eben auch rückwirkend nachgereicht werden.

> Und schön wäre es, wenn in der Artikel-Bearbeitung die Page-Views in einem einfachen Textfeld änderbar angezeigt würden.

Ich weiß, dass der Counter fraglich ist (Bots, etc.), aber dem Admin explizit zu erlauben, die beliebtesten Artikel zu beeinflussen – ist das wirklich wünschenswert?

**Comment by frase-git** [comment]
> aber dem Admin explizit zu erlauben, die beliebtesten Artikel zu beeinflussen – ist das wirklich wünschenswert?

Verbieten kann man es sowieso nicht. Wer will - kann.
Ich kam darauf, weil meine Tests schon etliche Pageviews erzeugt haben. Wenn ich mit einem Blog online gehen sollte, wäre es gut, mit Null zu starten. Das krieg ich aber hin.
Das Ganze sollten wir aber im Auge behalten und sehen, wie es in der späteren Praxis läuft.

**Comment by cmb69** [comment]
> Ich kam darauf, weil meine Tests schon etliche Pageviews erzeugt haben. Wenn ich mit einem Blog online gehen sollte, wäre es gut, mit Null zu starten.

Ah, guter Punkt! Da wäre vielleicht in der Pluginadministration ein "Artikel-Views zurücksetzen" angebracht.

> Das Ganze sollten wir aber im Auge behalten und sehen, wie es in der späteren Praxis läuft.

Ja, auf jeden Fall.

**Comment by cmb69** [comment]
Well, after reconsideration I think it's best to keep it simple, and just store the count (not individual visits), and to export/import the count for easy manipulation (if desired).

**Comment by cmb69** [comment]
Marking as BC break since the current count needs to be exported from a Realblog_XH 3 database, and imported into v4.
